### PR TITLE
Homework 6

### PR DESCRIPTION
**Pull request by OlgaPoddubnaia**

### CODE REVIEW

File: src/main/java/hw6/dataProvider/DataProviderForJson.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+public class DataProviderForJson {
+
+    public static MetalsAndColorsFormDataFromJson metalsAndColorsFormDataFromJson =
+            new MetalsAndColorsFormDataFromJson();
+
+    @DataProvider(name = "dataFromJson")
+    public static Object[][] dataFromJson() {
+        return new Object[][]{
+                {
+                        metalsAndColorsFormDataFromJson.getFromJsonObject_1().getSummary(),
```
Comment (troe88) [inline src/main/java/hw6/dataProvider/DataProviderForJson.java:15]: This is completely prohibited to create code with reference to amount of data. What if we have to increase data set ?

File: src/main/java/hw6/entities/MetalsAndColorsFormDataFromJson.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import lombok.Setter;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+public class MetalsAndColorsFormDataFromJson {
```
Comment (troe88) [inline src/main/java/hw6/entities/MetalsAndColorsFormDataFromJson.java:17]: Сurrent abstraction layer should not be care about source of this data, class needs to be renamed

File: src/main/java/hw6/pages/MetalsAndColorsPage.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+import com.epam.jdi.light.ui.html.elements.complex.RadioButtons;
+import org.openqa.selenium.WebElement;
+import org.testng.Assert;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+
+public class MetalsAndColorsPage extends WebPage {
```
Comment (troe88) [inline src/main/java/hw6/pages/MetalsAndColorsPage.java:21]: Where is MetalsAndColors Form ?
Take a look at Login form and Entity Driving Testing approach from the lecture.
My expectations is: Class -> Page -> Form::submit(entity)

**Review by troe88 — CHANGES_REQUESTED** [review]

File: src/main/java/hw6/dataProvider/DataProviderForJson.java (hunk 1 above)
Comment (OlgaPoddubnaia) [reply src/main/java/hw6/dataProvider/DataProviderForJson.java]: I changed the whole dataprovider concept

File: src/main/java/hw6/entities/MetalsAndColorsFormDataFromJson.java (hunk 2 above)
Comment (OlgaPoddubnaia) [reply src/main/java/hw6/entities/MetalsAndColorsFormDataFromJson.java]: deleted this layer

File: src/main/java/hw6/pages/MetalsAndColorsPage.java (hunk 3 above)
Comment (OlgaPoddubnaia) [reply src/main/java/hw6/pages/MetalsAndColorsPage.java:14]: I made a  MetalsAndcolorsForm

File: src/main/java/hw6/forms/MetalsAndColorsForm.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    @UI("[id=calculate-button]")
+    public Button calculateButton;
+
+    @UI("[id=submit-button]")
+    public Button submitButton;
+
+
+    @Override
+    public void fill(MetalsAndColors metalsAndColors) {
```
Comment (troe88) [inline src/main/java/hw6/forms/MetalsAndColorsForm.java:63]: This method is used for FILLING form, but you are going to SUBMIT it. 
In this casem you should override submit method

File: src/main/java/hw6/pages/MetalsAndColorsPage.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    protected MetalsAndColorsForm metalsAndColorsForm;
+
+    @FindBy(css = "ul.results  > li")
+    private List<WebElement> results;
+
+    public void checkMetalsAndColorsUrl() {
+        MetalsAndColorsPage.checkUrl("https://jdi-testing.github.io/jdi-light/metals-colors.html");
```
Comment (troe88) [inline src/main/java/hw6/pages/MetalsAndColorsPage.java:22]: We have Page::checkOpen() method, that allow us to perform this check.
Try it out please

File: src/test/java/hw6/tests/MetalsAndColorsTest.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Test(dataProvider = "getDataFromJson",
+            dataProviderClass = DataProviderFromJson.class)
+    public void MetalsAndColorsFormPageTests(MetalsAndColors metalsAndColors) {
+        open();
+        login(ROMAN);
+        homePage.checkUserLoggedIn(ROMAN);
+        openMetalsAndColorsPage();
+        checkMetalsAndColorsPageUrl();
+       fillMetalsAndColors(metalsAndColors);
```
Comment (troe88) [inline src/test/java/hw6/tests/MetalsAndColorsTest.java:20]: Code convention issues, reformat your code please

**Review by troe88 — CHANGES_REQUESTED** [review]

File: src/main/java/hw6/forms/MetalsAndColorsForm.java (hunk 4 above)
Comment (troe88) [reply src/main/java/hw6/forms/MetalsAndColorsForm.java:63]: test

**Review by troe88 — CHANGES_REQUESTED** [review]

File: src/test/java/hw6/tests/MetalsAndColorsTest.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+import org.testng.annotations.Test;
+
+import static hw6.JdiSite.*;
+import static hw6.entities.User.ROMAN;
+
+public class MetalsAndColorsTest extends AbstractBaseTest {
+
+    @Test(dataProvider = "dataFromJson",
+            dataProviderClass = DataProviderForJson.class)
+    public void isMetalsAndColorsFormFilled(String[] summary, String[] elements,
```
[the remark this answers is not in the document]
Comment (OlgaPoddubnaia) [reply src/test/java/hw6/tests/MetalsAndColorsTest.java]: changed method name to metalsAndColorsFormTest

File: src/main/java/hw6/forms/MetalsAndColorsForm.java (hunk 4 above)
Comment (OlgaPoddubnaia) [reply src/main/java/hw6/forms/MetalsAndColorsForm.java:63]: added submit method

File: src/main/java/hw6/pages/MetalsAndColorsPage.java (hunk 5 above)
Comment (OlgaPoddubnaia) [reply src/main/java/hw6/pages/MetalsAndColorsPage.java:22]: deleted this method and changed it on JdiSite to checkOpened()

File: src/test/java/hw6/tests/MetalsAndColorsTest.java (hunk 6 above)
Comment (OlgaPoddubnaia) [reply src/test/java/hw6/tests/MetalsAndColorsTest.java]: reformed code

**Review by troe88 — APPROVED** [review]

**Pull request merged**